services/catalog/src/simcore_service_catalog/models/domain/service.py

```diff
@@ -0,0 +1,277 @@
+import sys
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Type, Union
+
+from pydantic import BaseModel, EmailStr, Extra, Field, FilePath, HttpUrl, constr
+from pydantic.schema import schema
+from pydantic.types import ConstrainedInt, ConstrainedStr, PositiveInt, StrictBool
+
+current_file = Path(sys.argv[0] if __name__ == "__main__" else __file__).resolve()
+
+KEY_RE = r"^(simcore)/(services)/(comp|dynamic)(/[^\s]+)+$"
+VERSION_RE = r"^(0|[1-9]\d*)(\.(0|[1-9]\d*)){2}(-(0|[1-9]\d*|\d*[-a-zA-Z][-\da-zA-Z]*)(\.(0|[1-9]\d*|\d*[-a-zA-Z][-\da-zA-Z]*))*)?(\+[-\da-zA-Z]+(\.[-\da-zA-Z-]+)*)?$"
+PROPERTY_TYPE_RE = r"^(number|integer|boolean|string|data:([^/\s,]+/[^/\s,]+|\[[^/\s,]+/[^/\s,]+(,[^/\s]+/[^/,\s]+)*\]))$"
```

Comment:
TIP: make sure to add some unit tests with the examples on these regular expressions. It is not the first time these expressions are changed and break it all at runtime

Reply:
some tests were added.